x-pack/plugins/lists/server/routes/list_index/create_list_index_route.ts

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { validate } from '@kbn/securitysolution-io-ts-utils';
+import { transformError } from '@kbn/securitysolution-es-utils';
+import { LIST_INDEX } from '@kbn/securitysolution-list-constants';
+
+import { createListIndexResponse } from '../../../common/api';
+import type { ListsPluginRouter } from '../../types';
+import { buildSiemResponse, removeLegacyTemplatesIfExist } from '../utils';
+import { getListClient } from '..';
+
+export const createListIndexRoute = (router: ListsPluginRouter): void => {
+  router.post(
+    {
+      options: {
+        tags: ['access:lists-all'],
+      },
+      path: LIST_INDEX,
+      validate: false,
+    },
+    async (context, _, response) => {
+      const siemResponse = buildSiemResponse(response);
+
+      try {
+        const lists = await getListClient(context);
+
+        const listDataStreamExists = await lists.getListDataStreamExists();
+        const listItemDataStreamExists = await lists.getListItemDataStreamExists();
+
+        const templateListExists = await lists.getListTemplateExists();
+        const templateListItemsExists = await lists.getListItemTemplateExists();
+
+        if (!templateListExists || !listDataStreamExists) {
+          await lists.setListTemplate();
+        }
+
+        if (!templateListItemsExists || !listItemDataStreamExists) {
+          await lists.setListItemTemplate();
+        }
+
+        await removeLegacyTemplatesIfExist(lists);
+
+        if (listDataStreamExists && listItemDataStreamExists) {
```

Comment:
I don't know the reason it was done like this before, but I can guess, if there are any updates to templates, that would the only way to update them - by calling that API. Otherwise, if no updates were made to templates, that action won't have any effect